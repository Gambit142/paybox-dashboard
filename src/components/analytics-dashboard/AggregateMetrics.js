import PropTypes from 'prop-types';
import styles from './aggregate-metrics.module.css';

const AggregateMetrics = ({ aggregateMetricsData }) => (
  <section className="flex flex-column">
    {aggregateMetricsData.map((detail) => (
      <article key={detail.generalLabel} className={styles.metricsContainer}>
        <div className={`flex flex-column ${styles.metricsInfoDiv}`}>
          <div className={`flex ${styles.generalLabelDiv}`}>
            <span className={`sora-font-500 font-numeric light-blue fs-14 ${styles.textHeightSpacing1}`}>{detail.generalLabel}</span>
            <button type="button" className={`flex ${styles.moreBtn}`}>
              <span className={`nunito-font-500 font-numeric dark-blue ${styles.textHeightSpacing1}`}>More</span>
              <img src="./assets/images/dropdown-icon.svg" alt="dropdown-icon" />
            </button>
          </div>
          <div className={styles.timeIntervalDiv}>
            {detail.timeIntervalsMobile.map((timeInterval) => (
              <div key={timeInterval.label} className={`flex flex-column ${styles.infoDiv}`}>
                <span className={`light-blue font-numeric nunito-font-400 ${styles.textHeightSpacing1}`}>{timeInterval.label}</span>
                <span className={`dark-blue sora-font-600 font-numeric ${styles.textHeightSpacing2}`}>{timeInterval.value}</span>
              </div>
            ))}
          </div>
        </div>
      </article>
    ))}
  </section>
);

AggregateMetrics.displayName = 'AggregateMetrics';

AggregateMetrics.propTypes = {
  aggregateMetricsData: PropTypes.arrayOf(
    PropTypes.shape({
      generalLabel: PropTypes.string.isRequired,
      timeIntervalsMobile: PropTypes.arrayOf(
        PropTypes.shape({
          label: PropTypes.string.isRequired,
          value: PropTypes.string.isRequired,
        }),
      ).isRequired,
    }).isRequired,
  ).isRequired,
};

export default AggregateMetrics;
