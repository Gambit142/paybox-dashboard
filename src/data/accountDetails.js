const accountDetails = [
  {
    detailType: 'Total savings balance',
    detailAmount: '₦194,000',
  },
  {
    detailType: 'Total customers',
    detailAmount: '₦919',
  },
  {
    detailType: 'Total POS transactions',
    detailAmount: '0',
  },
];

export default accountDetails;
