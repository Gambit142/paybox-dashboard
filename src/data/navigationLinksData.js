import AnalyticsDashboard from '../pages/analyticsDashboard/AnalyticsDashboard';
import Personal from '../pages/personal/Personal';
import SaleHub from '../pages/sale-hub/SaleHub';
import TalentManagement from '../pages/talent-management/TalentManagement';
import SpendManagement from '../pages/spend-management/SpendManagement';
import Cards from '../pages/cards/Cards';
import Leaderboard from '../pages/leaderboard/Leaderboard';
import ProfileSettings from '../pages/profile-settings/ProfileSettings';
import BankPerformance from '../pages/bank-performance/BankPerformance';

const navigationLinksData = [
  {
    name: 'Home',
    icon: '/assets/images/home-icon.svg',
    path: '/',
    alt: 'home icon',
    component: AnalyticsDashboard,
  },
  {
    name: 'Analytics Dashboard',
    icon: '/assets/images/analytics-icon.svg',
    path: '/analytics_dashboard',
    alt: 'analytics icon',
    component: AnalyticsDashboard,
  },
  {
    name: 'Personal',
    icon: '/assets/images/strongbox.svg',
    path: '/personal',
    alt: 'strongbox icon',
    rightIcon: '/assets/images/Chevron-Right.svg',
    component: Personal,
  },
  {
    name: 'Sale Hub',
    icon: '/assets/images/shop.svg',
    path: '/sale_hub',
    alt: 'shop icon',
    rightIcon: '/assets/images/Chevron-Right.svg',
    component: SaleHub,
  },
  {
    name: 'Talent management',
    icon: '/assets/images/3dcube.svg',
    path: '/talent_management',
    alt: '3d cube icon',
    rightIcon: '/assets/images/Chevron-Right.svg',
    component: TalentManagement,
  },
  {
    name: 'Spend management',
    icon: '/assets/images/card-edit.svg',
    path: '/spend_management',
    alt: 'card edit icon',
    rightIcon: '/assets/images/Chevron-Right.svg',
    component: SpendManagement,
  },
  {
    name: 'Cards',
    icon: '/assets/images/cards.svg',
    path: '/cards',
    alt: 'cards icon',
    component: Cards,
  },
  {
    name: 'Leaderboard',
    icon: '/assets/images/medal-star.svg',
    path: '/leaderboard',
    alt: 'medal star icon',
    component: Leaderboard,
  },
  {
    name: 'Profile & Settings',
    icon: '/assets/images/security-user.svg',
    path: '/profile_settings',
    alt: 'security user icon',
    component: ProfileSettings,
  },
  {
    name: 'Bank performance',
    icon: '/assets/images/radar.svg',
    path: '/bank_performance',
    alt: 'radar icon',
    rightIcon: '/assets/images/Chevron-Right.svg',
    component: BankPerformance,
  },
];

export default navigationLinksData;
