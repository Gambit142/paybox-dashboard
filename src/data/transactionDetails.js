const twoMillion = '₦2,000,000';
const fiveMillion = '₦5,000,000';
const onePointTwoMillion = '₦1,200,000';

// Data for Aggregate Metrics for Two Weeks for Mobile version starts here

const timeIntervalsMobile = [
  {
    label: 'Today',
    value: onePointTwoMillion,
  },
  {
    label: 'Yesterday',
    value: onePointTwoMillion,
  },
  {
    label: 'This week',
    value: onePointTwoMillion,
  },
  {
    label: 'Last week',
    value: onePointTwoMillion,
  },
];

const aggregateMetricsForTwoWeeksMobile = [
  {
    generalLabel: 'Total transfers',
    timeIntervals: timeIntervalsMobile,
  },
  {
    generalLabel: 'Total inflows',
    timeIntervals: timeIntervalsMobile,
  },
  {
    generalLabel: 'Total transactions',
    timeIntervals: timeIntervalsMobile,
  },
  {
    generalLabel: 'Transaction count',
    timeIntervals: timeIntervalsMobile,
  },
];

// Data for Aggregate Metrics for Two Weeks for Mobile version ends here

// Data for Aggregate Metrics for Two Weeks for Desktop version starts here
const timeIntervalsDesktop = [
  {
    label: 'Today',
    value: twoMillion,
  },
  {
    label: 'Yesterday',
    value: fiveMillion,
  },
  {
    label: 'This week',
    value: fiveMillion,
  },
  {
    label: 'Last week',
    value: fiveMillion,
  },
];

// Data for Aggregate Metrics for Two Weeks for Desktop version ends here

const aggregateMetricsForTwoWeeksDesktop = [
  {
    generalLabel: 'Total transfers',
    timeIntervals: timeIntervalsDesktop,
  },
  {
    generalLabel: 'Total inflows',
    timeIntervals: timeIntervalsDesktop,
  },
  {
    generalLabel: 'Total transactions',
    timeIntervals: timeIntervalsDesktop,
  },
  {
    generalLabel: 'Transaction count',
    timeIntervals: timeIntervalsDesktop,
  },
];

// Data for Financial Metrics starts here

const financialMetrics = [
  {
    generalLabel: 'Total cashout',
    metric: [
      {
        label: 'Today',
        value: '₦1,200,000',
      },
      {
        label: 'Yesterday',
        value: '₦1,200,000',
      },
    ],
  },
  {
    generalLabel: 'Transaction status',
    metric: [
      {
        label: 'Successful',
        value: '422',
        color: 'green',
      },
      {
        label: 'Failed',
        value: '18',
        color: 'red',
      },
    ],
  },
  {
    generalLabel: 'Average Transaction Count',
    metric: [
      {
        label: 'Value',
        value: '₦20,400',
      },
      {
        label: 'Count',
        value: '422',
      },
    ],
  },
  {
    generalLabel: 'Total commission',
    metric: [
      {
        label: 'POS commission',
        value: '₦87,000',
      },
      {
        label: 'Lotto commission',
        value: '₦102,000',
      },
    ],
  },
  {
    generalLabel: 'Average commission per transaction',
    metric: [
      {
        label: 'POS commission',
        value: '₦500',
      },
      {
        label: 'Lotto commission',
        value: '₦200',
      },
    ],
  },
];

// Data for Financial Metrics ends here

export {
  aggregateMetricsForTwoWeeksDesktop,
  aggregateMetricsForTwoWeeksMobile,
  financialMetrics,
};
