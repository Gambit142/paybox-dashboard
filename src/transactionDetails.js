const basicInformationDetails = [
  {
  label: 'Total savings balance',
  value: '₦194,000'
  },
  {
    label: 'Total customers',
    value: '₦919'
  },
  {
    label: 'Total POS',
    value: '0'
  }
];

const twoMillion = '₦2,000,000';
const fiveMillion = '₦5,000,000';

const timeIntervals = [
  {
    label: 'Today',
    value: twoMillion
  },
  {
    label: 'Yesterday',
    value: fiveMillion
  },
  {
    label: 'This week',
    value: fiveMillion
  },
  {
    label: 'Last week',
    value: fiveMillion
  }
];

const aggregateMetricsForTwoWeeks = [
  {
    generalLabel: 'Total transfer',
    timeIntervals
  },
  {
    generalLabel: 'Total inflow',
    timeIntervals
  },
  {
    generalLabel: 'Total transactions',
    timeIntervals
  },
  {
    generalLabel: 'Total count',
    timeIntervals
  }
]

const fifnacialMetrics = [
  {
    generalLlabel: 'Total Cashout',
    metric: [
      {
        label: 'Today',
        value: '₦1,200,000'
      },
      {
        label: 'Yesterday',
        value: '₦1,200,000'
      }
    ]
  },
  {
    generalLlabel: 'Transaction status',
    metric: [
      {
        label: 'Successful',
        value: '422',
        color: 'green'
      },
      {
        label: 'Failed',
        value: '18',
        color: 'red'
      }
    ]
  },
  {
    generalLlabel: 'Average Transaction Count',
    metric: [
      {
        label: 'Value',
        value: '₦20,400'
      },
      {
        label: 'Count',
        value: '422'
      }
    ]
  },
  {
    generalLlabel: 'Total commission',
    metric: [
      {
        label: 'POS commission',
        value: '₦87,000'
      },
      {
        label: 'Lotto commission',
        value: '₦102,000'
      }
    ]
  },
  {
    generalLlabel: 'Average commission per transaction',
    metric: [
      {
        label: 'POS commission',
        value: '₦500'
      },
      {
        label: 'Lotto commission',
        value: '₦200'
      }
    ]
  }
]