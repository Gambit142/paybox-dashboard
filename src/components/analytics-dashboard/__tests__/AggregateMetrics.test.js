import { render, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import AggregateMetrics from '../AggregateMetrics';
import { aggregateMetricsForTwoWeeksDesktop } from '../../../data/transactionDetails';

describe('AggregateMetrics', () => {
  test('AggregateMetrics renders correctly', () => {
    const tree = renderer.create(<AggregateMetrics aggregateMetricsData={aggregateMetricsForTwoWeeksDesktop} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('AggregateMetrics renders aggregate metrics', async () => {
    render(<AggregateMetrics aggregateMetricsData={aggregateMetricsForTwoWeeksDesktop} />);
    await waitFor(() => {
      expect(screen.getByText('Total transfers')).toBeInTheDocument();
      expect(screen.getByText('Total inflows')).toBeInTheDocument();
      expect(screen.getByText('Total transactions')).toBeInTheDocument();
    });
  });
  test('AggregateMetrics renders time intervals', async () => {
    render(<AggregateMetrics aggregateMetricsData={aggregateMetricsForTwoWeeksDesktop} />);
    const elements = await screen.findAllByText('Today');
    expect(elements.length).toBe(4);
    expect(elements[0]).toBeInTheDocument();
  });
});