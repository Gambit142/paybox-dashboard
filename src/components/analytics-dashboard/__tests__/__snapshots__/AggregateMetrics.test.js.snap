// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateMetrics AggregateMetrics renders correctly 1`] = `
<section
  className="mainMetricsContainer"
>
  <article
    className="metricsContainer"
  >
    <div
      className="flex flex-column metricsInfoDiv"
    >
      <div
        className="flex generalLabelDiv"
      >
        <span
          className="sora-font-500 font-numeric light-blue fs-14 textHeightSpacing1"
        >
          Total transfers
        </span>
        <button
          className="flex moreBtn"
          type="button"
        >
          <span
            className="nunito-font-500 font-numeric dark-blue textHeightSpacing1"
          >
            More
          </span>
          <img
            alt="dropdown-icon"
            src="./assets/images/dropdown-icon.svg"
          />
        </button>
      </div>
      <div
        className="timeIntervalDiv"
      >
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Today
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦2,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Yesterday
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            This week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Last week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
      </div>
    </div>
  </article>
  <article
    className="metricsContainer"
  >
    <div
      className="flex flex-column metricsInfoDiv"
    >
      <div
        className="flex generalLabelDiv"
      >
        <span
          className="sora-font-500 font-numeric light-blue fs-14 textHeightSpacing1"
        >
          Total inflows
        </span>
        <button
          className="flex moreBtn"
          type="button"
        >
          <span
            className="nunito-font-500 font-numeric dark-blue textHeightSpacing1"
          >
            More
          </span>
          <img
            alt="dropdown-icon"
            src="./assets/images/dropdown-icon.svg"
          />
        </button>
      </div>
      <div
        className="timeIntervalDiv"
      >
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Today
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦2,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Yesterday
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            This week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Last week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
      </div>
    </div>
  </article>
  <article
    className="metricsContainer"
  >
    <div
      className="flex flex-column metricsInfoDiv"
    >
      <div
        className="flex generalLabelDiv"
      >
        <span
          className="sora-font-500 font-numeric light-blue fs-14 textHeightSpacing1"
        >
          Total transactions
        </span>
        <button
          className="flex moreBtn"
          type="button"
        >
          <span
            className="nunito-font-500 font-numeric dark-blue textHeightSpacing1"
          >
            More
          </span>
          <img
            alt="dropdown-icon"
            src="./assets/images/dropdown-icon.svg"
          />
        </button>
      </div>
      <div
        className="timeIntervalDiv"
      >
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Today
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦2,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Yesterday
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            This week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Last week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
      </div>
    </div>
  </article>
  <article
    className="metricsContainer"
  >
    <div
      className="flex flex-column metricsInfoDiv"
    >
      <div
        className="flex generalLabelDiv"
      >
        <span
          className="sora-font-500 font-numeric light-blue fs-14 textHeightSpacing1"
        >
          Transaction count
        </span>
        <button
          className="flex moreBtn"
          type="button"
        >
          <span
            className="nunito-font-500 font-numeric dark-blue textHeightSpacing1"
          >
            More
          </span>
          <img
            alt="dropdown-icon"
            src="./assets/images/dropdown-icon.svg"
          />
        </button>
      </div>
      <div
        className="timeIntervalDiv"
      >
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Today
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦2,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Yesterday
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            This week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
        <div
          className="flex flex-column infoDiv"
        >
          <span
            className="light-blue font-numeric nunito-font-400 textHeightSpacing1"
          >
            Last week
          </span>
          <span
            className="dark-blue sora-font-600 font-numeric textHeightSpacing2"
          >
            ₦5,000,000
          </span>
        </div>
      </div>
    </div>
  </article>
</section>
`;
